### DISCUSSION TITLE
2.0.49.3 release

### DISCUSSION THREAD

**Discussion by bizley** [yiisoft/yii2#20053] [General]
I've discussed some options to release 2.0.49.3 with only one thing being reverted without the stuff already merged (marked with 2.0.50 milestone) with @samdark and it's tricky because we have this subsplit repo to be updated from master. I don't have experience with that kind of flow so I need some help regarding handling it properly (it would be so easy without this subsplit).

I proposed something like:
- create backup branch for current master,
- create branch for 2.0.49.3 solving https://github.com/yiisoft/yii2/issues/20041
- switch this 2.0.49.3 branch to be a new master
- do the release of 2.0.49.3
- wait for subsplit update
- bring master back from backup

Samdark said it's too risky unfortunately.

**Comment by schmunk42** [comment]
Is it only possible to release from `master`?

> Samdark said it's too risky unfortunately.

Basically I agree with this.

**Comment by bizley** [comment]
I think we can release from any branch, not sure about it though. The problem is that the subsplit repo is updated from master automatically. At least this is what I understand.

**Comment by rob006** [comment]
Why we don't create 2.0.49.x branch and let subsplit script to handle it as any other branch? If 2.0.49 will be LTS, then we need this anyway, because there will be more releases like this with some important patches. It looks like it already worked for other branches in the past and 2.2.x line too: https://github.com/yiisoft/yii2-framework/branches

**Comment by rob006** [comment]
Also, I'm pretty sure this is not a first time when patch release is created not from master, but using old tag as a base. So there must be a way to do this without messing with `master`.

**Comment by bizley** [comment]
So just tag it from subsplit repo branch? I think this would work, just need a confirmation from @samdark

**Reply by terabytesoftw** [reply]
For the 2.2 branch, @cebe set up automatic updating.

**Reply by rob006** [reply]
No, I mean create 2.0.49.x branch in this repo, and configure subspliting to handle it the same way as 2.2 branch: https://github.com/yiisoft/yii2-framework/branches - subsplit repo does not seem to be tied to only one branch. I expect that tags created form this 2.0.49.x branch would be handled correctly in subsplit repo.

**Reply by samdark** [reply]
Should work according to @cebe

**Reply by bizley** [reply]
Let me know when and how can I proceed then, please.

**Reply by samdark** [reply]
1. Create `2.0.49.x` branch.
2. Tag release there via `./build/build` when the branch ↑ is checked out.
3. Wait for subsplit to work (it should). If there's a problem, contact @cebe.
4. Tag yii2-framework repo by checking out `2.0.49.x` branch.
5. Check if there's a proper tag at packagist.org yiisoft/yii2 etc.

**Reply by bizley** [reply]
Unfortunately I cannot see the new branch on subsplit. Contacted @cebe on Slack.

**Reply by cebe** [reply]
The [2.0.49.x](https://github.com/yiisoft/yii2-framework/tree/2.0.49.x) branch is now synced. @bizley you can create a tag now.

**Reply by cebe** [reply]
@samdark steps to reproduce:
1. on the bot server: cd /var/www/yii-subsplit
2. Add the Branch in `split.php`
3. Execute `./trigger-subsplit.sh`

I also created a README in that directory.

**Comment by bizley** [comment]
Hi guys, sorry for not responding, I've been on vacation.
Thanks @cebe , all works now

2.0.49.3 was just tagged.
